backend/service/bot/bot.go

```diff
@@ -0,0 +1,59 @@
+package bot
+
+import (
+	"strings"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/lyft/clutch/backend/service"
+	"github.com/lyft/clutch/backend/service/bot/slack"
+)
+
+const Name = "clutch.service.bot"
+
+func New(_ *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service, error) {
+	s := &svc{
+		logger: logger,
+		scope:  scope,
+	}
+	return s, nil
+}
+
+type svc struct {
+	logger *zap.Logger
+	scope  tally.Scope
+}
+
+type Service interface {
+	// Functions that perform provider-specific sanitization/formatting
+	// They all pass the command to the generic MatchCommand function
+	MatchSlackCommand(command string) string
```

Comment:
this is redundant on the interface, no need for a MatchSlack, MatchDiscord, etc. the service can be configured to sanitize appropriately in the call to MatchCommand.